pkg/component/delete/delete.go

```diff
@@ -24,6 +27,24 @@ func NewDeleteComponentClient(kubeClient kclient.ClientInterface) *DeleteCompone
 	}
 }
 
+func (o *DeleteComponentClient) ListKubernetesComponents(devfileObj parser.DevfileObj, path string) (list []unstructured.Unstructured, err error) {
+	components, err := devfileObj.Data.GetComponents(common.DevfileOptions{})
+	if err != nil {
+		return
+	}
+	var u unstructured.Unstructured
+	for _, component := range components {
+		if component.Kubernetes != nil {
+			u, err = service.GetK8sComponentAsUnstructured(component.Kubernetes, path, devfilefs.DefaultFs{})
```

Comment:
You could move the `GetK8sComponentAsUnstructured` function to some devfile-related package (libdevfile?) as it is 100% related to devfile